ArcGISRuntimeSDKQt_CppSamples/Analysis/FindServiceAreasForMultipleFacilities/FindServiceAreasForMultipleFacilities.cpp

```diff
@@ -0,0 +1,178 @@
+#ifdef PCH_BUILD
+#include "pch.hpp"
+#endif // PCH_BUILD
+
+#include "FindServiceAreasForMultipleFacilities.h"
+
+#include "FeatureLayer.h"
+#include "GraphicsOverlay.h"
+#include "Map.h"
+#include "MapQuickView.h"
+#include "PictureMarkerSymbol.h"
+#include "QueryParameters.h"
+#include "ServiceAreaParameters.h"
+#include "ServiceAreaPolygon.h"
+#include "ServiceAreaResult.h"
+#include "ServiceAreaTask.h"
+#include "ServiceFeatureTable.h"
+#include "SimpleFillSymbol.h"
+#include "SimpleRenderer.h"
+#include "TaskWatcher.h"
+
+#include <QUrl>
+
+using namespace Esri::ArcGISRuntime;
+
+namespace
+{
+const QUrl url("https://services2.arcgis.com/ZQgQTuoyBrtmoGdP/ArcGIS/rest/services/San_Diego_Facilities/FeatureServer/0");
+const QUrl imageUrl("https://static.arcgis.com/images/Symbols/SafetyHealth/Hospital.png");
+const QUrl serviceAreaTaskUrl("https://sampleserver6.arcgisonline.com/arcgis/rest/services/NetworkAnalysis/SanDiego/NAServer/ServiceArea");
+}
+
+FindServiceAreasForMultipleFacilities::FindServiceAreasForMultipleFacilities(QObject* parent /* = nullptr */):
+  QObject(parent),
+  m_map(new Map(Basemap::lightGrayCanvas(this), this))
+{
+  m_serviceAreasOverlay = new GraphicsOverlay(this);
+
+  // create fill symbols for rendering the results
+  m_fillSymbols.append(new SimpleFillSymbol(SimpleFillSymbolStyle::Solid, QColor(255, 166, 0, 66), this));
+  m_fillSymbols.append(new SimpleFillSymbol(SimpleFillSymbolStyle::Solid, QColor(255, 0, 0, 66), this));
+
+  m_facilitiesTable = new ServiceFeatureTable(url, this);
+  m_facilitiesFeatureLayer = new FeatureLayer(m_facilitiesTable, this);
+
+  // create a symbol to display the facilities
+  PictureMarkerSymbol* facilitiesSymbol = new PictureMarkerSymbol(imageUrl, this);
+  facilitiesSymbol->setWidth(25);
+  facilitiesSymbol->setHeight(25);
+  m_facilitiesFeatureLayer->setRenderer(new SimpleRenderer(facilitiesSymbol, this));
+
+  // add the facilities feature layer to the map
+  m_map->operationalLayers()->append(m_facilitiesFeatureLayer);
+}
+
+FindServiceAreasForMultipleFacilities::~FindServiceAreasForMultipleFacilities() = default;
+
+void FindServiceAreasForMultipleFacilities::init()
+{
+  // Register the map view for QML
+  qmlRegisterType<MapQuickView>("Esri.Samples", 1, 0, "MapView");
+  qmlRegisterType<FindServiceAreasForMultipleFacilities>("Esri.Samples", 1, 0, "FindServiceAreasForMultipleFacilitiesSample");
+}
+
+MapQuickView* FindServiceAreasForMultipleFacilities::mapView() const
+{
+  return m_mapView;
+}
+
+bool FindServiceAreasForMultipleFacilities::taskRunning() const
+{
+  return m_taskRunning;
+}
+
+// Set the view (created in QML)
+void FindServiceAreasForMultipleFacilities::setMapView(MapQuickView* mapView)
+{
+  if (!mapView || mapView == m_mapView)
+    return;
+
+  m_mapView = mapView;
+  m_mapView->setMap(m_map);
+
+  m_mapView->graphicsOverlays()->append(m_serviceAreasOverlay);
+
+  connect(m_facilitiesFeatureLayer, &FeatureLayer::doneLoading, this, [this](Error loadError){
```

Comment:
this connection needs to happen before you set the map on the mapView (otherwise there's a chance it will be loaded before you set up the connection).